demo/flutter/lib/main.dart

```diff
@@ -0,0 +1,413 @@
+                ? null
+                : Text(
+                    errorMessage,
+                    style: TextStyle(color: Colors.white, fontSize: 18),
+                  )));
+  }
+
+  Widget footer = Expanded(
+      flex: 1,
+      child: Container(
+          alignment: Alignment.bottomCenter,
+          padding: EdgeInsets.only(bottom: 20),
+          child: const Text(
+            "Made in Vancouver, Canada by Picovoice",
+            style: TextStyle(color: Color(0xff666666)),
+          )));
+}
+
+typedef RecordedCallback = Function(double length);
+typedef ProcessErrorCallback = Function(LeopardException error);
+
+class MicRecorder {
+  final VoiceProcessor? _voiceProcessor;
+  int _sampleRate;
+
+  final RecordedCallback _recordedCallback;
+  final ProcessErrorCallback _processErrorCallback;
+  RemoveListener? _removeVoiceProcessorListener;
+  RemoveListener? _removeErrorListener;
+
+  List<int> _pcmData = [];
+
+  static Future<MicRecorder> create(int sampleRate, RecordedCallback recordedCallback, ProcessErrorCallback processErrorCallback) async {
+    return MicRecorder._(sampleRate, recordedCallback, processErrorCallback);
+  }
+
+  MicRecorder._(this._sampleRate, this._recordedCallback, this._processErrorCallback)
+      : _voiceProcessor = VoiceProcessor.getVoiceProcessor(512, _sampleRate) {
+    if (_voiceProcessor == null) {
+      throw LeopardRuntimeException("flutter_voice_processor not available.");
+    }
+    _removeVoiceProcessorListener =
+        _voiceProcessor!.addListener((buffer) async {
+      List<int> frame;
+      try {
+        frame = (buffer as List<dynamic>).cast<int>();
+      } on Error {
+        LeopardException castError = LeopardException(
+            "flutter_voice_processor sent an unexpected data type.");
+        _processErrorCallback(castError);
+        return;
+      }
+
+      _pcmData.addAll(frame);
+      _recordedCallback(_pcmData.length / _sampleRate);
+    });
+
+    _removeErrorListener = _voiceProcessor!.addErrorListener((errorMsg) {
+      LeopardException nativeError = LeopardException(errorMsg as String);
+      _processErrorCallback(nativeError);
+    });
+  }
+
+  Future<void> startRecord() async {
+    if (_voiceProcessor == null) {
+      throw LeopardInvalidStateException(
+          "Cannot start Leopard - resources have already been released");
+    }
+
+    _pcmData.clear();
+
+    if (await _voiceProcessor?.hasRecordAudioPermission() ?? false) {
+      try {
+        await _voiceProcessor!.start();
+      } on PlatformException {
+        throw LeopardRuntimeException(
+            "Audio engine failed to start. Hardware may not be supported.");
+      }
+    } else {
+      throw LeopardRuntimeException(
+          "User did not give permission to record audio.");
+    }
+  }
+
+  Future<File> stopRecord() async {
+    if (_voiceProcessor == null) {
+      throw LeopardInvalidStateException(
+          "Cannot start Leopard - resources have already been released");
```

Comment:
Pedantic, but in this case we're just starting audio recording, not Leopard itself